test/functional/rest.py

```diff
@@ -252,6 +261,13 @@ def run_test(self):
         assert_greater_than(int(response.getheader('content-length')), 80)
         response_str = response.read()
 
+        # get binary block with range
+        response = http_get_call_with_headers(
+            url.hostname, url.port, '/rest/block/' + bb_hash + self.FORMAT_SEPARATOR + "bin", {'Range': '10-29'}, True)
```

Comment:
Is this compliant with the standard? I'm not certain, but by my reading of the RFC you must also include the units of the range being requested, which in this case would be `bytes=`.

https://www.rfc-editor.org/rfc/rfc9110.html#name-range-requests

Reply:
@richmills3 You're right. It should start with a unit like `bytes=`, I forgot when I implemented this.
MDN has an example too https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range

Reply:
Further reading of the spec (https://httpwg.org/specs/rfc9110.html#field.content-range) makes me also think the response code should be 206 in this case, do you agree?

I think for us to take this change I'd like to see:

1.  `bytes=` unit added to the request parsing.
2. Response code 206 for range responses (unless I'm wrong about that).
3. More tests, in particular some negative tests, eg; for range requests outside of block, bad request format.
4. Also see my comment on `block_index.cpp`.

Reply:
I totally agree, these changes are needed and great. 

I'm trying to deal with some issues with building the project. There are some strange errors during the `make`(https://github.com/Ljzn/bitcoin-sv/actions/runs/3800283694/jobs/6463746175).  Maybe I should rebase my branch to the latest code on the master. Anyway, I will start coding on these as long as I find some free time and solve the build problem.

Reply:
I think you're hitting some known build issues on later OSs (Ubuntu 22?). We have fixes for those that will be in the next release but I don't think they're in master yet. They are easy to fix though, just `#include <mutex>` as the compiler suggests.